### PR TITLE
Better json

### PR DESCRIPTION
**Pull request by alexdlaird**
This PR depends on the `critical` flag. Merge that PR first (if you approve).

### CODE REVIEW

**Review by codingjoe — CHANGES_REQUESTED** [review]
There is still loads of stuff in here – like critical or not – that doesn't belong here. I am also not so sure about the setting.
I would rather suggest to add a separate URL or GET parameter in addition to the header.

**Comment by alexdlaird** [comment]
@codingjoe FYI, probably going to revert most of this PR and just add a simple example (illustrating overriding the JSON method in the view, since you already provide that leverage) to the README instead. Sound good?

**Comment by codingjoe** [comment]
sounds good, maybe you want to wait until your other PR is merged?

**Pull request closed without merging**
